 camera: Fix high battery drain after using flashlight

After turning off the flashlight via the tile, the camera is kept open,
preventing the device from entering deep sleep (and destroying battery
life).

Properly close the camera after turning off the flashlight in order to fix
the high battery drain.

diff --git a/services/camera/libcameraservice/CameraFlashlight.cpp b/services/camera/libcameraservice/CameraFlashlight.cpp
index 74fc498..6169839 100644
--- a/services/camera/libcameraservice/CameraFlashlight.cpp
+++ b/services/camera/libcameraservice/CameraFlashlight.cpp
@@ -897,7 +897,7 @@ status_t CameraHardwareInterfaceFlashControl::disconnectCameraDevice() {
     }
     mDevice->setPreviewWindow(NULL);
     mDevice->release();
-    mDevice = NULL;
+    mDevice.clear();
 
     return OK;
 }
